### PR TITLE
Add troubleshooting docs

### LINKED ISSUE #235: Introduce troubleshooting documentation and reduce chance of slow connectivity problems
### Docs
- **Docs**: Describe how to diagnoze some problems (e.g. get logs, kubectl describe etc.) (something like https://capact.io/docs/operation/diagnostics)
   - how to get logs
   - Use config-exporter to dump the configuration for all bug reports
      - See how it is used in https://github.com/kubeshop/botkube/blob/main/cmd/cli/cmd/migrate.go
      - Guide user to run the job on K8s cluster and get the result

- **Docs**: Describe common problems (like https://capact.io/docs/operation/common-problems)
   - e.g. Botkube killed by Kubernetes when internet connectivity is too slow
   - ~Bot not invited to channel - probably it's already fixed by our CLI~
   - Donna and Maria had the problems
    - Encourage to join community slack in case someone is stuck and is looking for help

### Botkube Open Source
- **CLI**: Ensure we link to the troubleshooting docs in CLI installation failed message
   - Maybe replace "Installation failed" error message to something different? It's not always about installation, it could be a temporary problem with a pod.
 - **Agent**: Consider increasing thresholds for the probes as plugin download might take a while (e.g. to 50? 100?)
    - https://github.com/kubeshop/botkube/blob/main/helm/botkube/values.yaml#L949
    - https://github.com/kubeshop/botkube/blob/main/helm/botkube/values.yaml#L962
- **Goreleaser config for plugins:** release plugins in `tar.gz` archives
    - For local env keep binaries anyway
    - We use go-getter so unpacking that will be out of the box (just need to double check they are packed without additional nested directory)

## Reason

Currently we lack a troubleshooting-related guide, which helps with basic debugging. Sometimes on our community Slack we receive similar questions, and we could have a single place with most common issues.

### PR DESCRIPTION
**Pull request by mszostok**
## Description

Changes proposed in this pull request:

- Add operations docs

## Testing

- https://add-operations.botkube-docs.pages.dev/next/operation/common-problems
- https://add-operations.botkube-docs.pages.dev/next/operation/diagnostics

## Related issue(s)

Fix https://github.com/kubeshop/botkube-docs/issues/235

### CODE REVIEW

File: docs/operation/common-problems.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+**Debugging steps**
+
+- [Check Botkube agent version](diagnostics.mdx#agent-version).
+
+- [Check plugin repository version](diagnostics.mdx#check-configured-plugin-repositories).
+
+**Solution**
+
+In order to fix the problem, you need to make sure that the agent version is the same as the plugin repository version. For example, for agent image `ghcr.io/kubeshop/botkube:v1.4.0` you need to configure official plugin repository in version `v1.4.0`: `https://github.com/kubeshop/botkube/releases/download/v1.4.0/plugins-index.yaml`.
```
Comment (huseyinbabal) [inline docs/operation/common-problems.md:25]: Should we be sure about, version replacer replaces this version or will it be always `v1.4.0`?

File: docs/operation/common-problems.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+- global-upload-edge.slack.com
+- wss://wss-backup.slack.com
+- wss://wss-mobile.slack.com
+- wss://wss-primary.slack.com
+
+Visit [Slack official guide](https://slack.com/help/articles/360001603387-Manage-Slack-connection-issues#network-settings) for troubleshooting your Slack connection.
+
+### Cloud Slack
+
+Cloud Slack integration communicates via gRPC with the Botkube control-plane. In order to make this integration work, you need to allow access to api.botkube.io.
```
Comment (huseyinbabal) [inline docs/operation/common-problems.md:65]:
```suggestion
Cloud Slack integration communicates via gRPC with the Botkube control-plane. In order to make this integration work, you need to allow access to `api.botkube.io`.
```

**Review by huseyinbabal — APPROVED** [review]
ok with minor comments

File: docs/operation/common-problems.md (hunk 1 above)
Comment (mszostok) [reply docs/operation/common-problems.md:25]: This is just an example, IMO we can leave it as it is, no need to have it always updated 👍

**Pull request merged**